crates/solver/src/solver/http_solver.rs

```diff
@@ -516,6 +520,25 @@ impl Solver for HttpSolver {
             }
         };
 
+        // Upload the instance if configured. Happens in a task to not delay solving.
+        if let Some(uploader) = &self.instance_uploader {
```

Comment:
`HttpSolver::solve()` is already very big. This functionality could be moved into a separate method.